src/frontend/components/SubscribeScheduledVideo/index.tsx

```diff
@@ -105,27 +105,19 @@ export const SubscribeScheduledVideo = ({
     }
   }, []);
 
-  const hasUserInToken = () => {
-    return getDecodedJwt().user !== undefined;
-  };
-
-  const hasEmailInToken = () => {
-    return hasUserInToken() && getDecodedJwt().user!.email;
-  };
-
-  const canIcheckIfIamRegistered = () => {
+  const isLtiToken = () => {
     return (
-      hasUserInToken() &&
-      (getDecodedJwt().user!.email ||
-        (getDecodedJwt().user!.id && getDecodedJwt().context_id))
+      getDecodedJwt().context_id &&
+      getDecodedJwt().consumer_site &&
+      getDecodedJwt().user !== undefined &&
+      getDecodedJwt().user!.id
     );
   };
```

Comment:
You should assigne the decodedToken to a variebl and then work with it. Typescript will not warn you anymore about missing `user`:

```
const isLtiToken = () => {
    const decodedJWT = getDecodedJwt();
    return (
      decodedJWT.context_id &&
      decodedJWT.consumer_site &&
      decodedJWT.user &&
      decodedJWT.user.id
    );
  };
```